__tests__/app.test.js

```
@@ -148,4 +148,81 @@ describe("GET /api/reviews/:review_id/comments", () => {
+        return request(app)
+        .post("/api/reviews/10/comments")
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+            const returnedComment = response.body.comment
+            expect(returnedComment).not.toHaveProperty("propertyToBeIgnored")
+        })
+    })
+    it("404: Endpoint returns not found when a user tries to post a comment on a review that does not exist", () => {
+        const newComment = {username: "bainesface", body: "I am a comment to fulfill the post request."}
+        return request(app)
+        .post("/api/reviews/9999/comments")
+        .send(newComment)
+        .expect(404)
+        .then((response) => {
+            const errorMessage = response.body
+            expect(errorMessage).toEqual({status: "404", msg: "Not found."})
+        })
+    })
+    it("400: When a comment is passed to an invalid path, a bad request error is returned", () => {
+        const newComment = {username: "bainesface", body: "I am a comment to fulfill the post request."}
+        return request(app)
+        .post("/api/reviews/not-an-id/comments")
+        .send(newComment)
+        .expect(400)
+        .then((response) => {
+            const errorMessage = response.body
+            expect(errorMessage).toEqual({status: "400", msg: "Bad request."})
+        })
+    })
+    it("400: When passed a comment object that is missing a required property, a bad request error is returned", () => {
+        const newComment = {username: "bainesface"}
+        return request(app)
+        .post("/api/reviews/10/comments")
+        .send(newComment)
+        .expect(400)
+        .then((response) => {
+            const errorMessage = response.body
+            expect(errorMessage).toEqual({status: "400", msg: "Bad request."})
+        })
+    })
+    it("400: When a comment is attempted to be posted by an invalid username, a bad request error is sent", () => {
```

this should return a 404 error